src/pages/ListSwaps/SwapRow/SwapRow.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Button,
+  CircularProgress,
+  TableCell,
+  TableRow
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+import React, { useEffect, useReducer } from "react";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { EmbeddedRepresentationSubEntity } from "../../../../gen/siren";
+import executeAction from "../../../api/executeAction";
+import { Asset, Properties, toMainUnit } from "../../../api/swapTypes";
+import Dialog from "../../../components/Dialog";
+import ExternalLink from "../../../components/ExternalLink";
+import LedgerActionDialogBody from "../LedgerActionDialogBody";
+import SirenActionParametersDialogBody from "../SirenActionParametersDialogBody";
+import SwapStatusIcon from "../SwapStatusIcon";
+import {
+  actionButtonClicked,
+  actionFailed,
+  actionSuccessful,
+  closeLedgerActionDialog,
+  closeSirenParametersDialog,
+  sirenParameterDialogSubmitted
+} from "./events";
+import { ActionExecutionStatus, initialState, reducer } from "./reducer";
+
+interface AssetCellProps {
+  asset: Asset;
+}
+
+function AssetCell({ asset }: AssetCellProps) {
+  return <span>{toMainUnit(asset)}</span>;
+}
+
+const useStyles = makeStyles(() => ({
+  tableRow: {
+    cursor: "pointer"
+  }
+}));
+
+interface SwapRowProps extends RouteComponentProps {
+  swap: EmbeddedRepresentationSubEntity;
+  reload: () => void;
+}
+
+function SwapRow({ swap, history, reload }: SwapRowProps) {
+  const [
+    {
+      state: {
+        actionExecutionStatus,
+        activeLedgerActionDialog,
+        activeSirenParameterDialog
+      },
+      sideEffect
+    },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    if (!sideEffect) {
+      return;
+    }
+
+    switch (sideEffect.type) {
+      case "reloadData": {
+        setTimeout(() => reload(), 1000);
```

Comment:
:mega: 
Wait, why do we need this?

I don't expect this reload to be enough for ledger actions but it should do it for "Accept" and "Decline" without the timeout right?

Reply:
apated - removed timeout, added no-action fixture in tests that is returned after successful accept.

Reply:
>I don't expect this reload to be enough for ledger actions but it should do it for "Accept" and "Decline" without the timeout right?

It actually is big enough for ethereum regtest :stuck_out_tongue_winking_eye: 

For the accept and decline, I think the problem was that we weren't actually reloading after performing them. But it should be reloading after my changes.